cmd/object-api-input-checks.go

```diff
@@ -166,13 +166,25 @@ func checkObjectArgs(ctx context.Context, bucket, object string, obj ObjectLayer
 	if err := checkBucketExist(ctx, bucket, obj); err != nil {
 		return err
 	}
+
+	if err := validObjectNameLength(bucket, object); err != nil {
+		return err
+	}
+
+	if err := checkObjectNameForPrefixSlash(bucket, object); err != nil {
+		logger.LogIf(ctx, ObjectNamePrefixAsSlash{
```

Comment:
the returned error is actually same as the one constructed in logger.LogIf() it is a frivolous log, we can avoid since we are sending a proper XML back to the client. 

One more change needed in this PR is we can converge both the functions into one such as 
```
func checkValidObjectName(bucket, object) error {
```

Also use `IsValidObjectName()` inside this function such that all 3 different types of errors can be returned appropriately by a single function.

Reply:
`checkObjectArgs` calls `IsValidObjectName`  which checks for  trailing "/" in object name and returns error as minio does not support object names with trailing "/".  Whereas `checkPutObjectArgs` doesn't checks `IsValidObjectName` and allows the trailing slash. 
With this PR `object name length` and `name prefix as slash` check is added in both `checkObjectArgs`  and `checkPutObjectArgs`  . 

Thus `IsValidObjectName` can not be clubbed to `checkObjectNameForLengthAndSlash`   and kept seperate.

This PR also adds  test for object name length and name prefix as slash  in `TestPutObjectLongName` of `server_test.go` .